storage/innobase/row/row0merge.cc

```diff
@@ -1166,16 +1159,21 @@ dberr_t row_merge_buf_write(const row_merge_buf_t *buf,
 
 		if (blob_file) {
 			ut_ad(buf->index->is_primary());
-			err = row_merge_buf_blob(
-				entry, n_fields, &blob_heap, blob_file);
+			if (!blob_heap) {
+				blob_heap = mem_heap_create(100);
+			}
+
+			err = row_merge_buf_blob(entry->fields, n_fields,
+						 blob_file, buf->index,
```

Comment:
Nice: `blob_file` is the 3rd argument, which matches the parameters of this function `#ifdef DBUG_OFF`. That means that the compiler has a chance to pass through this parameter without having to insert an instructions for copying the parameter register to another one.